perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -408,6 +408,178 @@ public Group updateGroup(PerunSession sess, Group group) throws InternalErrorExc
 		return group;
 	}
 
+	public Group updateParentGroupId(PerunSession sess, Group group) throws InternalErrorException {
+		if(group == null) throw new InternalErrorException("Group can't be null.");
+
+		//return group with updated parentGroupId
+		group = getGroupsManagerImpl().updateParentGroupId(sess, group);
+
+		return group;
+	}
+
+	public void moveGroup(PerunSession sess, Group destinationGroup, Group movingGroup) throws InternalErrorException, GroupMoveNotAllowedException {
+
+		// check if moving group is null
+		if (movingGroup == null) {
+			throw new GroupMoveNotAllowedException("Moving group: " + movingGroup + " cannot be null.");
+		}
+
```

Comment:
Please add check, that "movingGroup" is NOT "Members" group too, like you do for destinationGroup.